lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -542,23 +575,39 @@ def _can_service_start(self) -> bool:
 
         return True
 
-    def _initialize_admin_user(self):
+    def _initialize_admin_user(self, pwd: Optional[str] = None):
         """Change default password of Admin user."""
-        hashed_pwd, pwd = generate_hashed_password()
-        self.secrets.put(Scope.APP, "admin_password", pwd)
+        is_update = pwd is not None
+        hashed_pwd, pwd = generate_hashed_password(pwd)
+
+        if is_update:
+            # TODO replace with user_manager.patch_user()
+            resp = self.opensearch.request(
+                "PATCH",
+                "/_plugins/_security/api/internalusers/admin",
+                [{"op": "replace", "path": "/hash", "value": hashed_pwd}],
+            )
+            logger.debug(f"Patch admin user response: {resp}")
+            if resp.get("status") != "OK":
+                raise OpenSearchError(f"{resp}")
+        else:
+            # reserved: False, prevents this resource from being update-protected from:
```

Comment:
If we're no longer reserving this user, should we [hide](https://opensearch.org/docs/latest/security/access-control/api/#reserved-and-hidden-resources) it?

Reply:
I do not think so, when we get to the opensearch dashboard, I believe it would be useful to provide the full transparent list of existing users in the cluster.